C++/source/searching/linear_search.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+    Linear Search
+    -------------
+    A searching algorithm that finds the position of a target value
+    within an array (sorted or otherwise).
+
+    Time complexity
+    --------------
+    O(N), where N is the number of elements in the array
+
+    Space Complexity
+    --------------
+    O(1)
+*/
+
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+int linearSearch(const int value, const vector<int> &vect) {
```

Comment:
It would be better if this function is generic, i.e. it works not only for integers but for all types.

See http://www.cplusplus.com/doc/oldtutorial/templates/ for an introduction on templates in C++.